src/sinks/util/http.rs

```diff
@@ -598,6 +599,145 @@ pub fn validate_headers(
     Ok(validated_headers)
 }
 
+/// Request type for use in the `Service` implementation of HTTP stream sinks.
```

Comment:
Yep, I agree with all of that. I had similar thoughts about phasing out the `HttpBatchService` and inlining it eventually, when working on the changes but failed to call that out in the PR description; thanks for raising it.

I can add the deprecation, and update the new sink guide as part of this PR.

Reply:
Regarding the http sink guide- I think the guide currently does a good job on explaining how implement `Service` , which I think has value because while HTTP sinks are common, it's often the case that new sinks won't be able to leverage the `HttpService`.

What if we update the sink guide to call out a warning about the fact that we are implementing it from scratch for the purpose of demonstration, but that new HTTP based sink components should most likely use the `HttpService`. What do you think?

Reply:
Yeah, I like that 👍🏻